From 976dacab92a4b4fe7dbb648716180a21bba0b803 Mon Sep 17 00:00:00 2001
From: Miroslaw Swiader <miroslaw.swiader@intel.com>
Date: Tue, 9 Feb 2016 13:28:46 +0100
Subject: [PATCH 3/5] usb: dwc3: suspend usb2 phy in initialization

According to Synopsys spec: 11.1.4.8 B-Device Flow

Change-Id: Ib6a68a278e3db73c694213057b69db58ed11744a
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-19752
Signed-off-by: Miroslaw Swiader <miroslaw.swiader@intel.com>
Reviewed-on: https://android.intel.com:443/468862
---
 drivers/usb/dwc3/core.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/usb/dwc3/core.c b/drivers/usb/dwc3/core.c
index 59e7612..a1cf639 100644
--- a/drivers/usb/dwc3/core.c
+++ b/drivers/usb/dwc3/core.c
@@ -78,6 +78,8 @@ void dwc3_set_mode(struct dwc3 *dwc, u32 mode)
 #define GUSB2PHYACC0_REGDATA(v)  (v & 0xFF)
 #define GUSB2PHYACC0_REGDATA_MASK  0xFF
 
+static void dwc3_suspend_usb2_phy(struct dwc3 *dwc, bool suspend);
+
 static int ulpi_read(struct dwc3 *dwc, u32 reg)
 {
 	u32 val32 = 0, count = 200;
@@ -551,6 +553,8 @@ static int dwc3_core_init(struct dwc3 *dwc)
 
 	dwc3_suspend_usb3_phy(dwc, true);
 
+	dwc3_suspend_usb2_phy(dwc, true);
+
 	return 0;
 
 err0:
-- 
1.9.1

